vertx-web/src/main/java/io/vertx/ext/web/templ/TemplateEngine.java

```diff
@@ -33,12 +33,17 @@
 public interface TemplateEngine {
 
   /**
-   * Render
+   * Render the template
+   * <p>
+   * <b>NOTE</b> if you call method directly (i.e. not using {@link io.vertx.ext.web.handler.TemplateHandler}) make sure
+   * that <i>templateFileName</i> is sanitized via {@link io.vertx.ext.web.impl.Utils#normalizePath(String)}
+   *
    * @param context  the routing context
-   * @param templateFileName  the template file name to use
+   * @param templateDirectory  the template directory to use
+   * @param templateFileName  the relative template file name to use
    * @param handler  the handler that will be called with a result containing the buffer or a failure.
    */
-  void render(RoutingContext context, String templateFileName, Handler<AsyncResult<Buffer>> handler);
+  void render(RoutingContext context, String templateDirectory, String templateFileName, Handler<AsyncResult<Buffer>> handler);
```

Comment:
This is a breaking change and only required for inheritance, I guess the best would be to keep the original method and add this one as a new one. Then the old one would purelly delegate to this one with a `null` as a path for example.

```java
default void render(RoutingContext context, String templateFileName, Handler<AsyncResult<Buffer>> handler) {
  render(context, null, handler);
}
```